src/client/Microsoft.Identity.Client/IConfidentialClientApplication.cs

```diff
@@ -24,6 +25,11 @@ public partial interface IConfidentialClientApplication : IClientApplicationBase
         /// </remarks>
         ITokenCache AppTokenCache { get; }
 
+        /// <summary>
+        /// The certificate used to create this <see cref="ConfidentialClientApplication"/>, if any.
+        /// </summary>
+        X509Certificate2 Certificate { get; }
```

Comment:
Do we want to give an explicit implementation? (as this is a change of a public interface ?

Reply:
Sadly this isn't a possibility in MSAL, as this is a C# 8 feature, and C# 8 is not available with ancient .net 45

Reply:
I'd rather hide MSAL 4.20 - 4.22 from nuget :)

Reply:
So maybe an extension method?
Otherwise shall we bump-up the version to 5.0 ?

Reply:
IConfidentialClient2 ?

Reply:
I think extension method is better...

Reply:
I'm not a fan of IConfidentialClient2.